soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -19,268 +21,301 @@ extern GlobalContext* gGlobalCtx;
-                packet->effectCategory = "speed";
-                packet->timeRemaining = 30000;
-            }
-            else if (packet->effectType == "no_z") {
-                packet->effectCategory = "no_z";
-                packet->timeRemaining = 30000;
-            }
-            else if (packet->effectType == "spawn_wallmaster" ||
-                packet->effectType == "spawn_arwing" ||
-                packet->effectType == "spawn_darklink" ||
-                packet->effectType == "spawn_stalfos" ||
-                packet->effectType == "spawn_wolfos" ||
-                packet->effectType == "spawn_freezard" ||
-                packet->effectType == "spawn_keese" ||
-                packet->effectType == "spawn_icekeese" ||
-                packet->effectType == "spawn_firekeese" ||
-                packet->effectType == "spawn_tektite" ||
-                packet->effectType == "spawn_likelike" ||
-                packet->effectType == "cucco_storm") {
-                packet->effectCategory = "spawn";
-            }
-            else {
-                packet->effectCategory = "none";
-                packet->timeRemaining = 0;
-            }
+    std::string jsonPayload = payload.dump();
+    SDLNet_TCP_Send(socket, jsonPayload.c_str(), jsonPayload.size() + 1);
+}
 
-            // Check if running effect is possible
-            EffectResult effectResult = ExecuteEffect(packet->effectType.c_str(), packet->effectValue, 1);
-            if (effectResult == EffectResult::Retry || effectResult == EffectResult::Failure) {
-                nlohmann::json dataSend;
-                dataSend["id"] = packet->packetId;
-                dataSend["type"] = 0;
-                dataSend["timeRemaining"] = packet->timeRemaining;
-                dataSend["status"] = effectResult;
-
-                std::string jsonResponse = dataSend.dump();
-                SDLNet_TCP_Send(tcpsock, jsonResponse.c_str(), jsonResponse.size() + 1);
-            } else {
-                bool anotherEffectOfCategoryActive = false;
-                for (CCPacket* pack : receivedCommands) {
-                    if (pack != packet && pack->effectCategory == packet->effectCategory && pack->packetId < packet->packetId) {
-                        anotherEffectOfCategoryActive = true;
-
-                        nlohmann::json dataSend;
-                        dataSend["id"] = packet->packetId;
-                        dataSend["type"] = 0;
-                        dataSend["timeRemaining"] = packet->timeRemaining;
-                        dataSend["status"] = EffectResult::Retry;
-
-                        std::string jsonResponse = dataSend.dump();
-                        SDLNet_TCP_Send(tcpsock, jsonResponse.c_str(), jsonResponse.size() + 1);
-
-                        break;
-                    }
-                }
+CrowdControl::Effect* CrowdControl::ParseMessage(char payload[512]) {
+    nlohmann::json dataReceived = nlohmann::json::parse(payload);
```

Comment:
Regarding the making `ParseMessage` return nullptr on error, this could now just be:
```suggestion
    nlohmann::json dataReceived = nlohmann::json::parse(payload, nullptr, false);
    if (dataReceived.is_discarded()) {
        SPDLOG_ERROR("Error parsing JSON");
        return nullptr;
    }
```
or wrapping it in a `try ... catch`
```suggestion
    nlohmann::json dataReceived;
    try {
        dataReceived = nlohmann::json::parse(payload);
    } catch (nlohmann::json::parse_error& e) {
        SPDLOG_ERROR("Error parsing JSON: {}", e.what());
        return nullptr;
    } 
```

Reply:
Went with the first one, thanks a lot for the suggestion!